kivy/uix/textinput.py

```diff
@@ -2482,9 +2482,8 @@ def keyboard_on_key_down(self, window, keycode, text, modifiers):
                     self.copy()
             elif key == 27:
                 self.focus = False
-            return True
```

Comment:
Wouldn't this now allow the things below to execute, such as tab-insertion and copy-paste, which is not supposed to happen in read-only mode?

Reply:
I actually asked for this thing on discord. My goal is to create a textInput of fixed size(say 10 rows). This would be the user friendly  logs window. Since logs can be 1000's of lines , the user should be able to scroll this textinput to see more lines and copy the text. Also they should be able to move inside the textInput using arrow keys. In my case I would actually need the text to readonly so that the users while moving through doesn't messes up the logs but should be able to copy them.

Reply:
```

class InfoArea(TextInput):
    def insert_text(self, substring, from_undo=False):
        return True

    def keyboard_on_key_down(self, window, keycode, text, modifiers):
        if keycode[0] != 8:
            return super().keyboard_on_key_down(window, keycode, text, modifiers)
        return True



```

This is the code that I required.

Reply:
@matham i did some manual testing after the change, and i couldn’t insert a tab, or paste using ctrl-v, maybe i should add some unittesting for good mesure, but it seemed fine to me.

Reply:
Hmm then that code is counterintuitive and it needs comments to explain what is going on and possibly a fix. Because upon a simple reading it looks like it should insert tabs, or that tab insertion line at the bottom and possibly the others doesn't do anything.

Also, when `_editable` is False, it could be because the widget is disabled or readonly and possibly other modes (keyboard related?). And I think we may want to interact with it only if readonly, but not when it's not editable for other reasons. And even so, I'm not sure people who set it to readonly always want it to be inter-actable? So it may need another option, but I'm not sure.

So you'll probably need instead something like:
```
if self.readonly and self.readonly_interactive:
    ...
    return True
elif not self._editable:
    ...
    return True
else:
    ...
```

Additionally, with your change, it seems like while you can interact it with partially, you can't fully interact with it. E.g. you can shift select, but ctrl-a doesn't work. So it'll probably need some code from under the `elif text and not is_interesting_key` condition extracted and moved under the `if self.readonly` condition. So it's clear what is happening and only specific to that condition.

Reply:
Hopefuly the logic should be clearer now, i extracted handling of commands and shortcuts to specific methods, shortcut being used only at the top, and checking itself if text is editable or not, only allowing a subset of operations if text is not editable.
I also discovered a cute little bug with inserting tab while having a selection, it kept the selection alive, while other insertions did not, behavior has been changed to delete selection in this situation like any other insertion.